sdk/spring-boot-starter/src/main/scala/kalix/spring/boot/KalixContainerInitializer.scala

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2021 Lightbend Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kalix.spring.boot
+
+import java.io.File
+
+import scala.concurrent.Future
+
+import com.typesafe.config.Config
+import com.typesafe.config.ConfigFactory
+import kalix.devmode.KalixProxyContainer
+import org.slf4j.LoggerFactory
+import org.springframework.context.ApplicationContextInitializer
+import org.springframework.context.ApplicationListener
+import org.springframework.context.ConfigurableApplicationContext
+import org.springframework.context.event.ContextClosedEvent
+
+class KalixContainerInitializer
+    extends ApplicationContextInitializer[ConfigurableApplicationContext]
+    with ApplicationListener[ContextClosedEvent] {
+
+  private val logger = LoggerFactory.getLogger(this.getClass)
+
+  private val config = {
+    val mainConfig = ConfigFactory.load()
+
+    val devConfigFile = new File(System.getProperty("user.dir"), "dev-mode.conf")
```

Comment:
This is a bit unorthodox. 

I'm looking for a file called `dev-mode.conf` at the root of the project, not under `src/main/resources`. This file won't be packaged, so the final jar won't have it and will just work as expected in prod. 

An alternative is to place it under `src/main/resources` and configure maven/sbt to filter it out when packaging. 

The behavior that I'm looking for is the following: 

* a user should not have to pass the file using a `-D` option.
* the application should behave the same if started with `mvn spring-boot:run` or if main function is run from inside the editor (not extra config needed).
* when packaging the application, this file should not be include and `KalixProxyContainer` should not be started.

Reply:
Btw, I tested with IntelliJ and with VSCode and in both cases the editor runs the main function from the root directory and the "trick" works as expected. 

When running a package application, it also work. Meaning, dev-mode is not enabled because there is not file. However, there is one catch though. 

If you package the jar and run is with `java -jar target/my-application.jar` it will find the `dev-mode.conf` file. Instead, we need to: `(cd target && java -jar my-application.jar )`.

Reply:
Let me check that I understand the problem. If we let the user put the file in `src/main/resources` and then is included in the jar. Then when deployed in Kubernetes it will again create a container. Am I right? 
If so, how about this? Could we create a boolean that gets the entire `dev-mode` to false when `mvn deploy`? Meaning, it's our maven plugin that takes care of setting this creation of the container in Kubernetes off. WDYT?

Reply:
For clarity, as I understand option 3, we won't though the file, but we would ensure that the container build have a -D flag added to it.

Reply:
I would prefer 2 or 3. Let's imagine a situation in that a user will build the image manually and hack the -D flag. What are the possible consequences?

Reply:
Option 3 is the one that we would use a `-D` property. So I guess you mean option 1 or 2. 

None of the solutions is waterproof. In option 2, we will need to tweak the build to exclude `dev-mode.conf` from final package. Users will be able to remove that tweak because it will be part of the pom.xml / build.sbt config. 

The consequence of packaging such a file in prod is that it will try to start a dev-mode proxy inside the user pod. That will probably fail, but the application (user function) will start anyway and the prod proxy will find it.

Reply:
Now that we said that, I think Option 1 is the most safe. 

If we explicitly look-up for `dev-mode.conf` at the root of the application, then a user will have to consciously build a docker image with such a file at the root of the application. Not impossible, but quite a conscious action. 

While removing some configs from pom.xml/build.sbt can happen by accident or inadvertence.

Reply:
My comment was more about that I'm not afraid too much of hacking a docker image (which as mentioned is possible in all 3 scenarios) I was just curious if there are any really serious consequences that I'm not aware of. 

I still prefer 2 or 3, putting config files in the root directory feels wrong.

Reply:
Since I don't have any merit argument against it, let's start with config in the root.